applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nonlinear_strain_energy_response_function.cpp

```diff
@@ -0,0 +1,385 @@
+        {
+            if( std::abs(response_gradient_0[i]) > tolerance )
+            {
+                KRATOS_WATCH(response_gradient_0[i])
+                rResponseGradient[0] = response_gradient_1[i] / response_gradient_0[i];
+                break;
+            }
+        }
+
+        mResponseGradient_0[rAdjointCondition.Id()] = response_gradient_1;
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Element& rAdjointElement,
+                                                const Variable<array_1d<double, 3>>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY
+
+        if (rSensitivityGradient.size() != 0)
+            rSensitivityGradient.resize(0, false);
+
+        KRATOS_CATCH("");
+    }
+
+
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Element& rAdjointElement,
+                                                const Variable<double>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+
+        if (rSensitivityGradient.size() != 0)
+            rSensitivityGradient.resize(0, false);
+
+        KRATOS_CATCH("");
+    }
+
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Condition& rAdjointCondition,
+                                                const Variable<double>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
+
+        if (rSensitivityGradient.size() != 0)
+            rSensitivityGradient.resize(0, false);
+
+
+
+        KRATOS_CATCH("");
+    }
+
+
+    // Calculates the derivate of the response function with respect to the design parameters
+    // Computation of \frac{1}{2} (u^T_i - u^T_i-1)  \cdot ( \frac{\partial f_{ext}_i}{\partial x} + frac{\partial f_{ext}_i-1}{\partial x})
+    void AdjointNonlinearStrainEnergyResponseFunction::CalculatePartialSensitivity(Condition& rAdjointCondition,
+                                                const Variable<array_1d<double, 3>>& rVariable,
+                                                const Matrix& rSensitivityMatrix,
+                                                Vector& rSensitivityGradient,
+                                                const ProcessInfo& rProcessInfo)
+    {
+        KRATOS_TRY;
```

Comment:
You should smash an error if `rVariable` is not `SHAPE`.

Reply:
I will add this to my ToDo list, I am doing it like that now as I am mainly interested in shape sensitivities.

Reply:
I add an error message for now, and in the near future there would be modified a bit to consider element properties sensitivities, it is already done on another branch.